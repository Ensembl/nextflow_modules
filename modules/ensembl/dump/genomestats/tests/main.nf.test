// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// nf-core modules test dump/genomestats
nextflow_process {

    name "Test Process DUMP_GENOMESTATS"
    script "../main.nf"
    process "DUMP_GENOMESTATS"

    tag "modules"
    tag "modules_nfcore"
    tag "dump"
    tag "dump/genomestats"

    test("Dump Genome statistics [stub]") {
        when {

                options "-stub"

                process {
                    """
                    input[0] = [[ test_id:'dummy_core_db_core_110_1', species:'Genus species']]
                    """
                }
            }

            then {

                assert snapshot(
                    process.out.core_stats.collect {
                        it.collect { it instanceof String && file(it).exists() ? file(it).name : it }
                        }
                    ).match("genome_stats")

                assertAll(
                    { assert process.success },
                    { assert snapshot(process.out).match() },
                    { assert process.out.core_stats.size() == 1 },
                    { assert process.out.core_stats != null },
                )
            }
    }
}