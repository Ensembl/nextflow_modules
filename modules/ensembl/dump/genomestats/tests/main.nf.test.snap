{
    "Dump Genome statistics [stub]": {
        "content": [
            {
                "0": [
                    [
                        [
                            {
                                "test_id": "dummy_core_db_core_110_1",
                                "species": "Genus species"
                            }
                        ],
                        "core_stats.json:md5,dc964a618276360f339d65dabd3e279a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3874d3d94b6b4756c287444ccd013e28"
                ],
                "core_stats": [
                    [
                        [
                            {
                                "test_id": "dummy_core_db_core_110_1",
                                "species": "Genus species"
                            }
                        ],
                        "core_stats.json:md5,dc964a618276360f339d65dabd3e279a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3874d3d94b6b4756c287444ccd013e28"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-06-17T11:26:56.649041557"
    },
    "genome_stats": {
        "content": [
            [
                [
                    [
                        {
                            "test_id": "dummy_core_db_core_110_1",
                            "species": "Genus species"
                        }
                    ],
                    "core_stats.json"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-12-02T17:39:27.332030943"
    }
}